apps/src/code-studio/hashEmail.js

```diff
@@ -4,15 +4,18 @@ import $ from 'jquery';
 var EMAIL_REGEX = /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i;
 
 module.exports = function hashEmail(options) {
-  // hash the email, if it is an email
+  // Hash the email, if it is an email.
   var email = $(options.email_selector).val().toLowerCase().trim();
   if (email !== '' && EMAIL_REGEX.test(email)) {
     var hashed_email = CryptoJS.MD5(email);
     $(options.hashed_email_selector).val(hashed_email);
 
-    // if age < 13, don't send the plaintext email
-    if (!options.age_selector || $(options.age_selector).val() < 13) {
-      $(options.email_selector).val('');
+    // Unless we want to deliberately skip the step of clearing the email.
+    if (!options.skip_clear_email) {
+      // If age < 13, don't send the plaintext email.
+      if (!options.age_selector || $(options.age_selector).val() < 13) {
+        $(options.email_selector).val('');
+      }
```

Comment:
Left this intact since it's used by a number of haml files and I wanted to focus this change on signup, but signup in particular uses the skip_clear_email option and takes care of considering by account type separately in this PR.  Will note larger work item to fix this for the other situations.